ocaml/xapi/xapi_vm.ml

```diff
@@ -505,6 +507,17 @@ let create ~__context ~name_label ~name_description
     ~nVRAM
   : API.ref_VM =
 
+  (* TODO: Error handling, but how? raise? or return null ref? or put default values? *)
+  if power_state = `Halted && suspend_VDI != Ref.null then begin
+    error "No suspend_VDI should be provided if VM created in `Halted state";
+  end
+  else if power_state = `Suspended && suspend_VDI = Ref.null then begin
+    error "VM created in `Suspended state needs a suspend_VDI";
+  end
+  else begin
+    error "Bad power state for VM creation";
+  end;
+
```

Comment:
I'd like some feedback here, I'm not sure what to do in this error cases, for now there's only log but I'm willing to change it accordingly to your comments.

Thanks

Reply:
you can throw a VM bad power state error - https://github.com/xapi-project/xen-api/blob/cdcc916a3ed5fd3040d1bdcc33bf4faa46292f48/ocaml/idl/datamodel_vm.ml#L353

you will need to make sure you add a 'changed' to the datamodel for this function

Reply:
Thanks for the feedback, will do!